prymer/api/picking.py

```diff
@@ -136,42 +139,60 @@ def build_primer_pairs(
         region_end = max(p.span.end for p in right_primers)
         bases = fasta.fetch(target.refname, region_start - 1, region_end)
 
+    # Each tuple is left_idx, right_idx, penalty, tm
+    pairings: list[Tuple[int, int, float, float]] = []
+
+    for i in range(0, len(left_primers)):
+        for j in range(0, len(right_primers)):
+            lp = left_primers[i]
+            rp = right_primers[j]
+
+            # Ignore pairings with amplicons too large
+            if rp.span.end - lp.span.start + 1 > amplicon_sizes.max:
+                continue
+
+            # Ignore pairings with amplicons too small
+            if rp.span.end - lp.span.start + 1 < amplicon_sizes.min:
+                continue
+
+            amp_mapping = Span(refname=target.refname, start=lp.span.start, end=rp.span.end)
+            amp_bases = bases[amp_mapping.start - region_start : amp_mapping.end - region_start + 1]
+            amp_tm = calculate_long_seq_tm(amp_bases)
+
+            if amp_tm < amplicon_tms.min or amp_tm > amplicon_tms.max:
+                continue
+
+            penalty = score(
+                left_primer=lp,
+                right_primer=rp,
+                amplicon=amp_mapping,
+                amplicon_tm=amp_tm,
+                amplicon_sizes=amplicon_sizes,
+                amplicon_tms=amplicon_tms,
+                weights=weights,
+            )
+
+            pairings.append((i, j, penalty, amp_tm))
+
+    pairings.sort(key=lambda tup: tup[2])
```

Comment:
**suggestion** Add a comment indicating what we're sorting by (or refactor to use a dataclass)

Reply:
**issue** With this change, the docstring is stale.

> Returns:
>    an iterator over all the valid primer pairs, **unsorted**